jwst/scripts/adjust_wcs.py

```diff
@@ -0,0 +1,320 @@
+        "-x",
+        "--expand",
+        action="store_true",
+        help="Expand wild cards in input file names (cannot be combined with "
+             "-f/--file)."
+    )
+
+    parser.add_argument(
+        "-w",
+        "--wcs",
+        action="store_true",
+        help="Save output WCS only to an ASDF file (ignored with -u/--update)."
+    )
+
+    group_wcs = parser.add_argument_group(
+        title='WCS adjustment parameters'
+    )
+    group_wcs.add_argument(
+        "-r",
+        "--ra_delta",
+        type=float,
+        default=0.0,
+        help="Delta RA (longitude), degrees."
+    )
+    group_wcs.add_argument(
+        "-d",
+        "--dec_delta",
+        type=float,
+        default=0.0,
+        help="Delta DEC (latitude), degrees."
+    )
+    group_wcs.add_argument(
+        "-o",
+        "--roll_delta",
+        type=float,
+        default=0.0,
+        help="Delta roll angle, degrees."
+    )
+    group_wcs.add_argument(
+        "-s",
+        "--scale_factor",
+        type=float,
+        default=1.0,
+        help="Scale factor."
+    )
+
+    parser.add_argument(
+        "-v",
+        "--version",
+        action="version",
+        version="v{0}".format(jwst.__version__)
+    )
+
+    options = parser.parse_args()
+
+    if options.file:
+        if options.expand:
+            parser.error(
+                "argument -x/--expand: not allowed with argument -f/--file"
+            )
+        if len(options.arg0) > 1:
+            parser.error(
+                "only one input file allowed with argument -f/--file"
+            )
+
+    wcs_pars = [
+        options.ra_delta,
+        options.dec_delta,
+        options.roll_delta,
+        options.scale_factor
+    ]
+
+    if wcs_pars == [0.0, 0.0, 0.0, 1.0]:
+        logger.warning(
+            "All WCS adjustment parameters (delta RA, DEC, roll, and scale "
+            "factor) have default values (identical transformation)."
+        )
+        logger.warning("Nothing to do. Quitting.")
+        return
+
+    if options.expand:
+        files = []
+        for f in options.arg0:
+            files.extend(glob.glob(f))
+    else:
+        files = options.arg0
+
+    # check that all input files are of supported types:
+    try:
+        ftypes = [_file_type(f) for f in files]
+    except Exception as e:
+        logger.error(f"{e}")
+        logger.error("Script execution aborted.")
+        return
+
+    if 0 in ftypes:
+        logger.error(f"Input file '{files[ftypes.index(0)]}' is neither "
+                     "an ASDF nor a FITS data file.")
+        logger.error("Script execution aborted.")
+        return
+
+    for fname, ftype in zip(files, ftypes):
+        data_model = jwst.datamodels.open(fname)
+
+        if jwst.datamodels.JwstDataModel == type(data_model):
+            is_data_model = False
+            asdf_file = asdf.open(fname, mode='rw' if options.update else 'r')
+            in_wcs = asdf_file['wcs']
```

Comment:
I'm not sure I completely follow - if `asdf_file` is supposed to be e.g. a `cal.asdf` file opened with asdf, then the WCS object is in `asdf_file['meta']['wcs']. I may be misreading this.

Reply:
Both issues above are related and maybe there is a better solution/approach. The issue is how can I know where a WCS object in an `asdf` file is stored? In all JWST imaging data models, it is in `meta.wcs` (or `['meta']['wcs']`) but if saving a WCS manually in a "simple" `asdf` file (as opposite to a full data model). If trying to open a non-data-model file returns a `jwst.datamodels.JwstDataModel` then it is clear this is not a `ImageModel` or some other product created by the pipeline => it must be a "simple asdf" in which user stored just the WCS object.

I thought that maybe if an user saves just the WCS object to an `asdf` file then we could agree to save it in `['wcs']` key (which should have been documented, BTW). It seems counter-intuitive to require that WCS object be written at `['meta']['wcs']` for a non-data-model `asdf` file. But maybe we should require that WCS object be stored in `['meta']['wcs']` or maybe anywhere and have another parameter which could be used to indicate the location of the WCS object in an asdf file.

Reply:
> The issue is how can I know where a WCS object in an asdf file is stored? 

If you are talking about the user provided asdf file with a wcs object, I think we require that it is in the root of the tree, i.e. `asdf_file.tree['wcs']` or `asdf_file['wcs']`. We also require that the input file with the wcs is an asdf file and assume so. We are not allowing arbitrary files and location of the WCS but defining how the file should look like.

Reply:
So, `if jwst.datamodels.JwstDataModel == type(data_model):` is essentially to distinguish between user-created asdf file and a valid `DataModel` (or its subclass)

Reply:
> I think we require that it is in the root of the tree, i.e. asdf_file.tree['wcs'] or asdf_file['wcs']. We also require that the input file with the wcs is an asdf file and assume so.

Do you mean that this script should not support updating the WCS directly on a data model file? That is, should users extract the WCS from a data model, write it to an ASDF file, run this script, extract updated WCS and update the WCS in the data model and write the updated data model file? It think it would be much more convenient if we actually support data models directly as inputs.